components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/LoginContextManagementUtil.java

```diff
@@ -323,14 +325,43 @@ private static boolean isStepHasMultiOption(AuthenticationContext context) {
         return stepHasMultiOption;
     }
 
-    private static boolean isValidRequest(HttpServletRequest request, AuthenticationContext context) {
+    private static boolean canHandleAuthenticator(AuthenticationContext context, String authenticators) {
+
+        List<String> authenticatorsList = new ArrayList();
+        if (authenticators != null) {
+            String[] authenticatorIdPMappings = authenticators.split(";");
+            for (String authenticatorIdPMapping : authenticatorIdPMappings) {
+                String[] authenticatorIdPMapArr = authenticatorIdPMapping.split(":");
+                authenticatorsList.add(authenticatorIdPMapArr[0]);
+            }
+        }
+
+        Map<Integer, StepConfig> stepMap = context.getSequenceConfig().getStepMap();
+        if (stepMap != null && !stepMap.isEmpty()) {
```

Comment:
Shall we use CollectionUtils.isEmpty here?

Reply:
updated with `MapUtils.isNotEmpty()`